apps/full-site-editing/full-site-editing-plugin/global-styles/src/dom-updater.js

```diff
@@ -14,23 +15,27 @@ export default ( options, getOptionValue ) => {
 	domReady( () => {
 		const current = {};
 		const cssVariables = {};
+		const styleElement = document.createElement( 'style' );
+		document.body.appendChild( styleElement );
 		options.forEach( option => {
 			current[ option ] = null;
 			cssVariables[ option ] = `--${ option.replace( '_', '-' ) }`;
 		} );
 
 		subscribe( () => {
+			let styleProps = '';
 			Object.keys( current ).forEach( key => {
```

Comment:
As it currently stands, this is recalculating the vars and updating the style node every time any redux action is triggered. How about something like:

- If no option value has changed => bail early of the subscribe callback.
- If some option value has changed => update the style node with all the new option values, including resetting the style node content when `styleProps` is empty, as that's the result of option values being null/undefined which we need to account for as well.

Reply:
I just pushed a couple of updates, let me know if those address your concerns.